File: apps/auto-wechat/frontend/src/api/wechatPreview.ts
import { request } from './request'

export interface CreateWeChatPreviewPayload {
  title?: string
  bodyHtml?: string
  runId?: string
  layoutTemplateId?: string
}

export interface WeChatPreviewSession {
  token: string
  previewUrl: string
  expiresIn: number
  localOnly: boolean
  localHint?: string
}

export async function createWeChatPreviewSession(
  payload: CreateWeChatPreviewPayload,
): Promise<WeChatPreviewSession> {
  return request.post<WeChatPreviewSession>('/preview/wechat-sessions', payload)
}

export function wechatPreviewQrImageUrl(previewUrl: string, size = 220): string {
  const params = new URLSearchParams({
    size: `${size}x${size}`,
    margin: '8',
    data: previewUrl,
  })
  return `https://api.qrserver.com/v1/create-qr-code/?${params.toString()}`
}

File: apps/auto-wechat/backend/internal/interface/http/dto/wechat_preview.go
package dto

import "github.com/auto-wechat-tech/backend/internal/application"

type CreateWeChatPreviewRequest struct {
	Title              string `json:"title"`
	BodyHTML           string `json:"bodyHtml"`
	RunID              string `json:"runId"`
	LayoutTemplateID   string `json:"layoutTemplateId"`
}

type WeChatPreviewSessionResponse struct {
	Token      string `json:"token"`
	PreviewURL string `json:"previewUrl"`
	ExpiresIn  int    `json:"expiresIn"`
	LocalOnly  bool   `json:"localOnly"`
	LocalHint  string `json:"localHint,omitempty"`
}

func ToWeChatPreviewSessionResponse(session application.WeChatPreviewSession) WeChatPreviewSessionResponse {
	return WeChatPreviewSessionResponse{
		Token:      session.Token,
		PreviewURL: session.PreviewURL,
		ExpiresIn:  session.ExpiresIn,
		LocalOnly:  session.LocalOnly,
		LocalHint:  session.LocalHint,
	}
}
